tests/deposit_reclaim_test.cairo

```diff
@@ -0,0 +1,285 @@
+use core::num::traits::zero::Zero;
+use starknet_bridge::bridge::token_bridge::TokenBridge::__member_module_token_settings::InternalContractMemberStateTrait;
+use core::array::ArrayTrait;
+use core::serde::Serde;
+use core::result::ResultTrait;
+use core::option::OptionTrait;
+use core::traits::TryInto;
+use snforge_std as snf;
+use snforge_std::{
+    ContractClassTrait, EventSpy, EventSpyTrait, EventsFilterTrait, EventSpyAssertionsTrait
+};
+use starknet::{ContractAddress, storage::StorageMemberAccessTrait};
+use starknet_bridge::mocks::{
+    messaging::{IMockMessagingDispatcherTrait, IMockMessagingDispatcher}, erc20::ERC20
+};
+use piltover::messaging::{IMessaging, IMessagingDispatcher, IMessagingDispatcherTrait};
+use starknet_bridge::bridge::{
+    ITokenBridge, ITokenBridgeAdmin, ITokenBridgeDispatcher, ITokenBridgeDispatcherTrait,
+    ITokenBridgeAdminDispatcher, ITokenBridgeAdminDispatcherTrait, IWithdrawalLimitStatusDispatcher,
+    IWithdrawalLimitStatusDispatcherTrait, TokenBridge, TokenBridge::Event,
+    types::{TokenStatus, TokenSettings}
+};
+use openzeppelin::access::ownable::{
+    OwnableComponent, OwnableComponent::Event as OwnableEvent,
+    interface::{IOwnableTwoStepDispatcher, IOwnableTwoStepDispatcherTrait}
+};
+
+use openzeppelin::token::erc20::interface::{IERC20, IERC20Dispatcher, IERC20DispatcherTrait};
+use starknet::contract_address::{contract_address_const};
+use super::constants::{OWNER, L3_BRIDGE_ADDRESS, DELAY_TIME};
+use super::setup::{
+    deploy_erc20, deploy_token_bridge_with_messaging, deploy_token_bridge, enroll_token_and_settle
+};
+
+#[test]
+fn deposit_reclaim_ok() {
+    let (token_bridge, mut spy, messaging_mock) = deploy_token_bridge_with_messaging();
+    let usdc_address = deploy_erc20("usdc", "usdc");
+    let usdc = IERC20Dispatcher { contract_address: usdc_address };
+
+    enroll_token_and_settle(token_bridge, messaging_mock, usdc_address);
+
+    usdc.approve(token_bridge.contract_address, 100);
+    token_bridge.deposit(usdc_address, 100, snf::test_address());
```

Comment:
based on the deposit_ok PR, I think we can move this to a common function. but we can merge this one into deposit_ok first and then you can fix all things there?

Reply:
Yes makes sense, can be done in #6